test/unit/spec/nbextensions/appCell2/widgets/appCellWidget-spec.js

```diff
@@ -535,15 +548,70 @@ define([
                 it('can validate params on startup and be ready to run', async function () {
                     // add a valid param so the cell starts up ready to run
                     setUpCellInState(this, 'EDITING_COMPLETE');
+                    spyOn(this.appCellWidgetInstance.jobManager, '_initJob').and.callThrough();
                     await startRunningCell(this);
                     expect(this.appCellWidgetInstance.__fsm().getCurrentState().state).toEqual(
                         fsmState.EDITING_COMPLETE
                     );
+                    expect(this.appCellWidgetInstance.jobManager._initJob).not.toHaveBeenCalled();
                     const runButton = this.kbaseNode.querySelector(selectors.run);
                     expect(runButton).not.toHaveClass('hidden');
                     expect(runButton).not.toHaveClass('disabled');
                 });
 
+                it('restores the state of an existing job', async function () {
```

Comment:
Add in some extra tests to ensure that the cell Does The Right Thing when it starts up with various job states.